mesonbuild/modules/python.py

```diff
@@ -461,6 +478,65 @@ def get_variable_method(self, args, kwargs):
 
         return var
 
+    @permittedKwargs({'required', 'version'})
+    def find_package(self, args: str, kwargs: dict):
+        required: bool = kwargs.pop('required',True)
+        if not isinstance(required, bool):
+            raise MesonException(f"Expected a bool instead found {type(required)} for required parameter.")        
+        
+        want_version: str = kwargs.pop('version','')
+        if not isinstance(want_version, str):
+            raise MesonException(f"Expected a string instead found {type(want_version)} for version parameter.")
+        
+        want_package = args[0]
+        if not isinstance(want_package, str):
+            raise MesonException(f"Expected a string instead found {type(want_package)}")
+
+        command_importlib = textwrap.dedent("""\
+            try:
+                # only 3.8+
+                from importlib.metadata import version
+            except (ImportError,ModuleNotFoundError):
+                # try using backport
+                from importlib_metadata import version
+                # fail if it is not available
```

Comment:
Deferred until meson requires python 3.8, meson.build features cannot depend on thirdparty modules.

Reply:
I think even when meson requires python 3.8+, meson would be supporting installing on old python version (say py2), do you think this wait is needed?

Reply:
Python 3.7 is EOL on 2023-06-27 so likely not until around then...

Reply:
Hmm, then can `meson.build` feature depend on the python version it is on? What I am saying is make `.find_package()` work only when host python `>=3.8` or else it will simply fail.

Reply:
also, if this pr is going to wait till `2023-06-27`, I would be closing it(It would be a lot better to implement it that time rather than having this).